code/modules/flufftext/Hallucination.dm

```diff
@@ -626,24 +727,44 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		door_number = rand(0,4) //if 0 bolts all visible doors
 	var/count = 0
 	feedback_details += "Door amount: [door_number]"
+
 	for(var/obj/machinery/door/airlock/A in range(7, target))
 		if(count>door_number && door_number>0)
 			break
 		if(!A.density)
 			continue
 		count++
-		var/obj/effect/hallucination/fake_door_lock/lock = new(get_turf(A))
-		lock.target = target
-		lock.airlock = A
-		locks += lock
-		lock.lock()
-		sleep(rand(4,12))
-	sleep(100)
-	for(var/obj/effect/hallucination/fake_door_lock/lock in locks)
-		locks -= lock
-		lock.unlock()
-		sleep(rand(4,12))
-	qdel(src)
+		LAZYADD(airlocks_to_hit, A)
+
+	addtimer(VARSET_CALLBACK(src, unlocking, TRUE), 10 SECONDS)
+	START_PROCESSING(SSfastprocess, src)
```

Comment:
What happens if SSfastprocess slows down and the timer ss doesn't? You're playing with fire here

Reply:
I think the worst case scenario is:

- Hallucination wants 3 doors locked
- `process` runs, locks a door
- Fast process slows down, takes longer than 10 seconds
- The timer runs, making doors unlock
- The one locked door unlocks

Meaning, the worst case scenario is just less doors locking.

Reply:
Why put yourself in the position at all? you know when the doors will be locked, it's when the list of to be locked doors is empty, why not put the timer there? at least then you can be sure that you're not causing collisions. Better yet why not sure the process itself handle the delay, so things slow down in sync.

Reply:
I can do that, was mostly for the sake of code complexity.